File: backend/src/main/java/com/example/SCDProiectv2/Models/DeliveryPackage.java
package com.example.SCDProiectv2.Models;

import jakarta.persistence.*;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.Set;

@Entity
@Data
@Table(name = "Package")
public class DeliveryPackage {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "courier_id")
    private Courier courier;

    @Column(name = "created_on", nullable = false, updatable = false)
    private LocalDateTime createdOn = LocalDateTime.now();

    @Column(name = "delivery_address", nullable = false)
    private String deliveryAddress;

    @Column(name = "email", nullable = false)
    private String email;

    @Column(name = "phone_number", nullable = false)
    private String phoneNumber;

    @Column(name = "pay_on_delivery")
    private boolean payOnDelivery = false;

    @Enumerated(EnumType.STRING)
    @Column(name = "status")
    private PackageStatus status;

    @Column(name = "awb_number", nullable = false, unique = true)
    private String awbNumber;

    @Override
    public String toString() {
        return "DeliveryPackage{" +
                "id=" + id +
                ", deliveryAddress='" + deliveryAddress + '\'' +
                ", email='" + email + '\'' +
                ", phoneNumber='" + phoneNumber + '\'' +
                '}';
    }
}

File: desktop management UI/ManagementUI_CourierApp/ManagementUI_CourierApp/Models/DeliveryPackage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ManagementUI_CourierApp.Models
{
    public class DeliveryPackage
    {
        //public int id { get; set; }

        public DateTime? createdOn {  get; set; }
        public string deliveryAddress { get; set; }
        public string email { get; set; }
        public string phoneNumber { get; set; }
        public bool payOnDelivery { get; set; }
        public string status { get; set; }

        public Courier courier { get; set; }

        //public string status { get; set; }
        //public PackageStatus status { get; set; }
        //public bool payOnDelivery { get; set; }
        //public DateTime? DeliveryDate { get; set; }
    }
}
